python/test/unit/language/test_conversions.py

```diff
@@ -333,13 +333,17 @@ def test_typeconvert_downcast(src_dtype, dst_dtype, rounding, max_repr, device):
     if src_dtype != 'float32' and is_cuda() and torch.cuda.get_device_capability(0) < (9, 0):
         pytest.skip("non-float32 downcast tests only supported on NVGPU with compute capability 9.0+")
 
-    if dst_dtype in ('float8e5', 'float8e4nv') and rounding == 'rtne' and (is_hip() or torch.cuda.is_available() and torch.cuda.get_device_capability(0) < (9, 0)):
+    is_fp16_to_fp8e4nv_on_hip_mi300 = is_hip_mi300() and (src_dtype == 'float16') and \
```

Comment:
This skip logic is fairly complicated now. Can you clean it up a bit by separating cuda and hip and organize better?

Reply:
also is the test actually enabled on mi300? I still see "float8e4nv downcast with RTNE rounding tests only supported on NVGPU with compute capability 9.0+" in https://github.com/triton-lang/triton/actions/runs/12821954051/job/35788517217?pr=5513 ?

Reply:
> also is the test actually enabled on mi300? I still see "float8e4nv downcast with RTNE rounding tests only supported on NVGPU with compute capability 9.0+" in https://github.com/triton-lang/triton/actions/runs/12821954051/job/35788517217?pr=5513 ?

That's weird. It's not passed locally. I'll look into it.